### DISCUSSION TITLE
Invoke-PnPSiteTemplate - "Access is Denied" error

### DISCUSSION THREAD

**Discussion by davida1992** [pnp/powershell#4014] [Q&A] [answered]
So I've outputted a .pnp file by using the following command.

```powershell
Get-PnPSiteTemplate -Out "c:\temp2\sp\PnP-Provisioning-FileV2.pnp"
``` 
The file outputs correctly without errors as far as I can see.

I then use the **Invoke-PnPSiteTemplate** to connect to my new site.

I am then applying the template by using the following command:

```powershell
Invoke-PnPSiteTemplate -Path "c:\temp2\sp\PnP-Provisioning-FileV2.pnp" -ClearNavigation
``` 

It seems to be doing something for a good 2 or 3 minutes, and then I am met with the following error every time:

```powershell
Invoke-PnPSiteTemplate: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
```

If I go to the site, it seems to have applied the majority of the template, though I notice it hasn't applied the colour scheme, and the only image on the home page that should be there, is not actually there, it looks like a broken image. I'm not sure if these things would cause the access denied error, or if there is a more detailed log area I can view to try and determine what it doesn't like?

Any help is much appreciated.

**Comment by BaronSparky** [comment, marked as answer]
I have seen this sort of thing before on sites with `NoScript` enabled. You could try disabling that first then reapply the template.

Disable NoScript : `Set-PnPTenantSite -Url '<New site url>' -DenyAddAndCustomizePages:$false`

Debug info might show the issue also: `Set-PnPTraceLog -On -Level Debug`

Then try invoking your template.
